conans/model/info.py

```diff
@@ -26,10 +23,25 @@ def __init__(self, value_str, indirect=False):
         self.full_package_id = pref.id
 
         # sha values
-        if indirect:
-            self.unrelated_mode()
+        if package_id_mode:
+            try:
+                getattr(self, package_id_mode)()
+            except AttributeError:
+                raise ConanException("'%s' is not a known package_id_mode" % package_id_mode)
         else:
-            self.semver()
+            if indirect:
+                self.unrelated_mode()
+            else:
+                self.semver_mode()
+
```

Comment:
The code above could be simplified if we receive always the mode. Shouldn't be more "semver" defaulted here but as config level.

Reply:
Please check the comment above. The current behavior is different for direct and indirect requires, and I am not sure we want to keep that with the new mode.